vault/kv_helpers.go

```diff
@@ -78,6 +78,27 @@ func kvReadRequest(client *api.Client, path string, params map[string]string) (*
 	return api.ParseSecret(resp.Body)
 }
 
+func kvListRequest(client *api.Client, path string) ([]interface{}, error) {
+	log.Printf("[DEBUG] Listing secrets at %s from Vault", path)
+	resp, err := client.Logical().List(path)
+	if err != nil {
+		return nil, fmt.Errorf("error listing from Vault at path %s, err=%s", path, err)
+	}
+
+	if resp == nil {
+		return nil, fmt.Errorf("no secrets found at %q", path)
+	}
+
+	// Return key names if they are present in response
+	if keyNameList, ok := resp.Data["keys"]; ok && keyNameList != nil {
+		if keyNames, ok := keyNameList.([]interface{}); ok {
```

Comment:
If you want to check the type assert failure then I think you should return an error when `!ok`, since this means that the value for `keys` is not in the form that we expect it to be.

Reply:
Fixed in 107c2da